packages/service-worker/worker/src/driver.ts

```
@@ -773,6 +792,17 @@ export class Driver implements Debuggable, UpdateSource {
           throw new Error(`Invariant violated (assignVersion): latestHash was null`);
         }
 
+        if (isWorkerScriptRequest) {
+          if (!this.clientVersionMap.has(event.resultingClientId)) {
+            // New worker hasn't been seen before; set this client to latest hash as well
+            this.clientVersionMap.set(event.resultingClientId, this.latestHash);
+          } else if (this.clientVersionMap.get(event.resultingClientId)! !== this.latestHash) {
+            throw new Error(
+              `Version mismatch between worker client ${event.resultingClientId} and requesting client ${clientId}`,
+            );
```

I'm not sure if we should throw an error, or fallback to network via `return null;`